public/app/features/alerting/unified/components/rules/RuleDetailsMatchingInstances.tsx

```diff
@@ -20,11 +20,13 @@ import { isAlertingRule } from '../../utils/rules';
 import { DetailsField } from '../DetailsField';
 
 import { AlertInstancesTable } from './AlertInstancesTable';
+import { getComponentsFromStats } from './RuleStats';
 
 interface Props {
   rule: CombinedRule;
   pagination?: PaginationProps;
   itemsDisplayLimit?: number;
+  enableFiltering?: boolean;
```

Comment:
I'm not entirely sure if I like this pattern – this is indicative to me that we need two separate components here.

```jsx
function () {
  // some state management here for filters and such
  const [filters, setFilters] = useState()

  // derive filtered instances from the filters
  const filteredInstances = useMemo(() => instances.filter(() => {  }), [filters])

  return (
    <>
      <InstancesFilters />
      <InstancesList />
    </>
  )
}
```

In fact, wherever we're using `<RuleDetailsMatchingInstances enableFiltering={false} />` it might make more sense to just use `<AlertInstancesTable />`?

Reply:
The `enableFiltering` prop is a cheap solution to the problem we cannot filter instances when limits are applied on the alert list view. I agree this is not the best way to express the intention, it was the easiest way to implement, though.
Unfortunately, we cannot just use `<AlertInstancesTable />`. There is a bit of code responsible for showing/hiding the table footer when there are more items available than we display. The refactoring is not so trivial in this case I'm afraid
We could try to change the `<AlertInstancesTable />`. However, it is also used by the alert list panel.